recipes/gnuradio/meta.yaml

```
@@ -0,0 +1,275 @@
+{% set version = "3.8.0.0" %}
+
+package:
+  name: gnuradio
+  version: {{ version }}
+
+source:
+  url: https://github.com/gnuradio/gnuradio/releases/download/v{{ version }}/gnuradio-{{ version }}.tar.gz
+  sha256: 3a9c90111f22f2f6f30450731dc671ad28ce824fc1a7bc8ea0783da9b95a7092
+  patches:
+    - 0001-Use-raw-strings-for-paths-substituted-directly-into-.patch
+    - 0001-Don-t-use-hard-coded-path-constants.patch
+    - 0001-python-Do-not-link-against-python-lib-for-building-a.patch
+
+build:
+  number: 0
+  skip: True  # [win and py2k]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake >=3.8
+    - ninja
+    - pkg-config  # [not win]
+   # libudev needed to link against libusb on linux
+    - {{ cdt('libudev') }}  # [linux]
+   # below are needed to link with Qt for qtgui
+    - {{ cdt('libice') }}  # [linux]
+    - {{ cdt('libselinux') }}  # [linux]
+    - {{ cdt('libsm') }}  # [linux]
+    - {{ cdt('libx11') }}  # [linux]
+    - {{ cdt('libxau') }}  # [linux]
+    - {{ cdt('libxcb') }}  # [linux]
+    - {{ cdt('libxdamage') }}  # [linux]
+    - {{ cdt('libxext') }}  # [linux]
+    - {{ cdt('libxfixes') }}  # [linux]
+    - {{ cdt('libxrender') }}  # [linux]
+    - {{ cdt('libxxf86vm') }}  # [linux]
+    - {{ cdt('libuuid') }}  # [linux]
+    - {{ cdt('mesa-dri-drivers') }}  # [linux]
+    - {{ cdt('mesa-libgl-devel') }}  # [linux]
+
+  host:
+    - boost-cpp
+    - click
+    - click-plugins
+    - fftw
+    - gsl
+    # blas needed to link with gsl
+    - libblas
+    - libcblas
+    - libusb >=1
+    - log4cpp
+    - mako
+    - mpir
+    # we don't build against numpy C lib
+    # (only needed at runtime, except gnuradio CMake checks for it at build)
+    # specify version so pin is ignored
+    - numpy >=1.1
```

It's better to take the global pin:
```suggestion
    - numpy
```
See https://github.com/conda-forge/root-feedstock/pull/68#issuecomment-558266527

---

This is a hack that I found necessary to get it to build. First, numpy is only actually a runtime dependency and the C API is not used at all. However, the build system checks that numpy is there at build time in order to ensure that runtime dependency, so in practice it becomes a host dependency. So why not just accept the pin then? Well maybe it's a bug that should be fixed another way, but I found that ``.circleci/build_steps.sh`` sets ``export CONDA_NPY='19'``, and even though the comment in the script says that the value is unused, that was causing a version conflict between the numpy 1.9 that was specified and the numpy pulled in by some of the dependencies that required the version from the conda-forge pin (currently 1.14?). So this was a hack to get the build to ignore the pin since the package doesn't care about the C API anyway.

---

If you don't use the C API then this is fine 👍